indico/modules/categories/models/categories.py

```diff
@@ -335,6 +335,65 @@ def parent_chain_query(self):
         """
         return self._get_chain_query(Category.id == self.parent_id)
 
+    def nth_parent(self, n_categs, fail_on_overflow=True):
+        """Return the nth parent of the category.
+
+        :param n_categs: the number of categories to go up
+        :param fail_on_overflow: whether to fail if we try to go above the root category
+                                 (or provide a negative ``n_categs``)
+        :returns: ``Category`` object or None (only if ``fail_on_overflow`` is not set)
+        """
+        if n_categs == 0:
+            return self
+        chain = self.parent_chain_query.all()
+        if n_categs < 0:
+            if fail_on_overflow:
+                raise OverflowError("Number of categories can't be negative!")
+            else:
+                return None
+        if n_categs > len(chain):
+            if fail_on_overflow:
+                raise OverflowError("Root category has no parent!")
+            else:
+                return None
+        return chain[::-1][n_categs - 1]
+
+    def is_child_of(self, categ):
+        return categ in self.parent_chain_query.all()
```

Comment:
`categ != self and self.parent_chain_query.filter(Category.id == categ.id).count():`